plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/HiveMigrations.java

```diff
@@ -0,0 +1,144 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.procedure;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.log.Logger;
+import io.trino.filesystem.FileEntry;
+import io.trino.filesystem.FileIterator;
+import io.trino.filesystem.Location;
+import io.trino.filesystem.TrinoFileSystem;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.parquet.ParquetDataSource;
+import io.trino.parquet.ParquetReaderOptions;
+import io.trino.parquet.metadata.ParquetMetadata;
+import io.trino.parquet.reader.MetadataReader;
+import io.trino.plugin.hive.FileFormatDataSourceStats;
+import io.trino.plugin.hive.HiveStorageFormat;
+import io.trino.plugin.hive.parquet.TrinoParquetDataSource;
+import io.trino.plugin.iceberg.fileio.ForwardingInputFile;
+import io.trino.plugin.iceberg.util.OrcMetrics;
+import io.trino.plugin.iceberg.util.ParquetUtil;
+import io.trino.spi.TrinoException;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DataFiles;
+import org.apache.iceberg.Metrics;
+import org.apache.iceberg.MetricsConfig;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.avro.Avro;
+import org.apache.iceberg.mapping.MappingUtil;
+import org.apache.iceberg.mapping.NameMapping;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.google.common.base.Verify.verify;
+import static io.trino.spi.StandardErrorCode.NOT_SUPPORTED;
+
+public final class HiveMigrations
+{
+    private static final Logger log = Logger.get(HiveMigrations.class);
+
+    private static final MetricsConfig METRICS_CONFIG = MetricsConfig.getDefault();
+
+    public enum RecursiveDirectory
+    {
+        TRUE,
+        FALSE,
+        FAIL,
+        /**/
+    }
+
+    private HiveMigrations() {}
+
+    public static List<DataFile> buildDataFiles(
+            TrinoFileSystem fileSystem,
+            RecursiveDirectory recursive,
+            HiveStorageFormat format,
+            String location,
+            PartitionSpec partitionSpec,
+            Optional<StructLike> partition,
+            Schema schema)
+            throws IOException
+    {
+        // TODO: Introduce parallelism
+        FileIterator files = fileSystem.listFiles(Location.of(location));
+        ImmutableList.Builder<DataFile> dataFilesBuilder = ImmutableList.builder();
+        while (files.hasNext()) {
+            FileEntry file = files.next();
+            String fileLocation = file.location().toString();
+            String relativePath = fileLocation.substring(location.length());
+            if (relativePath.contains("/_") || relativePath.contains("/.")) {
```

Comment:
*question*: what do `/_` and `/.` strings mean in this context?

Reply:
They're hidden directories in Hive world. You take a look at [HiveFileIterator](https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/fs/HiveFileIterator.java).